integrations/tensorflow/bindings/python/pyiree/tf/support/tf_test_utils.py

```diff
@@ -488,39 +746,35 @@ def setUp(self):
         module.create_reinitialized() for module in self._tar_modules
     ]
 
-  def compare_backends(self, trace_function, rtol=1e-6, atol=1e-6):
+  def compare_backends(self, trace_function):
     """Run the reference and target backends on trace_function and compare them.
 
     Args:
       trace_function: a function accepting a TracedModule as its argument.
-      rtol: a float as defined in np.allclose. Will be overridden in any
-        comparison between calls to the trace with a `rtol` kwarg.
-      atol: a float as defined in np.allclose. Will be overridden in any
-        comparison between calls to the trace with a `atol` kwarg.
     """
     # Trace the test function for each backend.
-    ref_trace = tf_utils.TracedModule(self._ref_module, trace_function)
+    ref_trace = TracedModule(self._ref_module, trace_function)
```

Comment:
It doesn't make sense to me that trace_function would be an argument to TracedModule. It seems preferable to be something like:

```
# Trace name is required argument to the traced.
ref_trace_mod = TracedModule(self._ref_module, trace_name=trace_function.__name__)
trace_function(ref_trace_mod)
ref_trace = ref_trace_mod.snapshot_trace()
```

Reply:
Do you think it's worth making the returned trace a snapshot (deepcopy) of the trace on the `TracedModule`?